commons/src/main/java/io/aiven/kafka/connect/common/config/AivenCommonConfig.java

```diff
@@ -16,217 +16,35 @@
-                if (longValue < 0) {
-                    throw new ConfigException(name, value, "Value must be at least 0");
-                } else if (longValue > maximumBackoffPolicy) {
-                    throw new ConfigException(name, value,
-                            "Value must be no more than " + maximumBackoffPolicy + " (24 hours)");
-                }
-            }
-        }, ConfigDef.Importance.MEDIUM,
-                "The retry backoff in milliseconds. "
-                        + "This config is used to notify Kafka Connect to retry delivering a message batch or "
-                        + "performing recovery in case of transient exceptions. Maximum value is "
-                        + TimeUnit.HOURS.toMillis(24) + " (24 hours).",
-                GROUP_RETRY_BACKOFF_POLICY, 1, ConfigDef.Width.NONE, KAFKA_RETRY_BACKOFF_MS_CONFIG);
-    }
-
-    public Long getKafkaRetryBackoffMs() {
-        return getLong(KAFKA_RETRY_BACKOFF_MS_CONFIG);
+        SinkCommonConfig.addKafkaBackoffPolicy(configDef);
     }
 
     protected static void addOutputFieldsFormatConfigGroup(final ConfigDef configDef,
             final OutputFieldType defaultFieldType) {
-        int formatGroupCounter = 0;
-
-        addFormatTypeConfig(configDef, formatGroupCounter);
-
-        configDef.define(FORMAT_OUTPUT_FIELDS_CONFIG, ConfigDef.Type.LIST,
-                Objects.isNull(defaultFieldType) ? null : defaultFieldType.name, // NOPMD NullAssignment
-                new OutputFieldsValidator(), ConfigDef.Importance.MEDIUM,
-                "Fields to put into output files. " + "The supported values are: " + OutputField.SUPPORTED_OUTPUT_FIELDS
-                        + ".",
-                GROUP_FORMAT, formatGroupCounter++, ConfigDef.Width.NONE, FORMAT_OUTPUT_FIELDS_CONFIG,
-                FixedSetRecommender.ofSupportedValues(OutputFieldType.names()));
-
-        configDef.define(FORMAT_OUTPUT_FIELDS_VALUE_ENCODING_CONFIG, ConfigDef.Type.STRING,
-                OutputFieldEncodingType.BASE64.name, new OutputFieldsEncodingValidator(), ConfigDef.Importance.MEDIUM,
-                "The type of encoding for the value field. " + "The supported values are: "
-                        + OutputFieldEncodingType.SUPPORTED_FIELD_ENCODING_TYPES + ".",
-                GROUP_FORMAT, formatGroupCounter++, ConfigDef.Width.NONE, FORMAT_OUTPUT_FIELDS_VALUE_ENCODING_CONFIG,
-                FixedSetRecommender.ofSupportedValues(OutputFieldEncodingType.names()));
-
-        configDef.define(FORMAT_OUTPUT_ENVELOPE_CONFIG, ConfigDef.Type.BOOLEAN, true, ConfigDef.Importance.MEDIUM,
-                "Whether to enable envelope for entries with single field.", GROUP_FORMAT, formatGroupCounter,
-                ConfigDef.Width.SHORT, FORMAT_OUTPUT_ENVELOPE_CONFIG);
+        SinkCommonConfig.addOutputFieldsFormatConfigGroup(configDef, defaultFieldType);
     }
 
     protected static void addFormatTypeConfig(final ConfigDef configDef, final int formatGroupCounter) {
```

Comment:
this method has no dependencies. May be we can delete this

Reply:
These are actually called by the GCS and Azure Sink configs at the moment, in the future we will be able to move them to point to the common sink but at the moment this is still required.

Reply:
Well this PR splits out the config, it is not supposed to update the GCS Sink connector and the Azure source connector.
I am trying to limit scope creep and keep the PR as manageable a size as poossible.

Reply:
Agree, but we have already deleted/moved several methods from AivenCommonConfig, leaving behind this method which is unused. Couldn't understand why.

Reply:
One of these methods i think is not actively being called by the GCS/Azure sink connectors, that one I just need to make sure is not used before I delete it. 
But the remaining methods need to be there until we do a major release and we can remove the deprecated class.